services/src/datasets/in_memory.rs

```diff
@@ -152,6 +155,13 @@ impl HashMapStorable for GdalMetaDataStatic {
     }
 }
 
+impl HashMapStorable for GdalMetadataNetCdfCf {
```

Comment:
we should define one Enum for all the GdalMetadataTypes and use that here. If we have to add all the concrete types here  there is no benefit in using a generic type anyway.

Reply:
I guess you are right that this is not an elegant implementation but this is out of scope for this task. You could add a new "idea" into our board if you have something in mind.